iceoryx_posh/test/fuzztests/README.md

```diff
@@ -0,0 +1,49 @@
+
+The fuzz wrappers currently support three different interfaces: the unix domain socket communication, the message parsing method and the toml configuration file parser.  
+1. The unix domain socket can either be tested by starting RouDi and sending messages to RouDi via uds or by directly invoking the message parsing function within RouDi. The first solution has the advantage, that the messages from uds to RouDi take the same path as normal messages from applications would do. The drawback however is that RouDi needs to be started with all the overhead which would not be necessary for this test which therefore slows the test a little bit.
+
+2. By invoking the message parsing function directly, the method is independent from the underlying protocol such as uds. It should also be slightly faster since some functions are not invoked compared to uds fuzzing. However, a RouDi thread is also started with this approach because otherwise it wouldn't be possible to invoke the message process function within RouDi without directly modifying the code in RouDi. 
+
+3. As a third use-case, the toml configuration parser can be tested. An example of a toml file can be found here: \[path\_to\_iceoryx\]/iceoryx\_posh/etc/iceoryx/roudi\_config\_example.toml. If you chose this method, you also need to set -t or --toml-file <PATH\_TO\_FILE>. A file needs to be specified which can be used to write in the messages which will send to the toml configuration parser. 
+
+The interface you want to fuzz can be chosen with -f or --fuzzing-api + entering the argument: uds, com or toml. The argument uds is used for use-case 1, com for use-case 2 and toml for use-case 3. 
+
+The way how the fuzzer sends messages to the interface can also be specified via -m or --input-mode
+1. The messages can be sent via stdin (-m stdin)
+2. The messages can be sent via using the command line (-m cl) followed by -i or --command-line-input with the message following as argument or -c --command-line-file to specify a file where the message you want to send is located. By using -c or --command-line-file, the complete file is sent as one message and is not sent line by line. 
+
+Additionally, the log level can be chosen via -l or --log-level such that messages are printed for debug or not. The default is to switch logging off. 3 arguments are possible:
+1. off: to disable logging (standard case)
+2. fatal: log fatal messages
+3. debug: log debug messages
+
+As an example: afl can be started for uds with: afl-fuzz -i \[path\_to\_input\_example\] -o \[path\_where\_output\_of\_fuzzer\_is\_written\_to\] \[path\_to\_iceoryx\]/build/posh/test/posh\_fuzztests -f uds -m stdin
```

Comment:
A more concrete example would be nice, something like:
```bash
cd build
afl-fuzz -i ../iceoryx_posh/test/fuzztests/fuzz_input/ -o fuzzer_output posh/test/posh_fuzztests -f uds -m stdin
```
Maybe a short description what the user can find in the output folder and what is useful to look at.

Reply:
Done